### PR TITLE
Adds avro integration

### PR DESCRIPTION
**Pull request by AdamM-FC**
This PR adds avro integration so all messages to and from kafka are encoded and decoded with avro

### CODE REVIEW

File: app/controllers/application_controller.rb [hunk 1]
```diff
@@ -17,9 +17,9 @@ def serialize_object_with_links(object, serializer)
     }
   end
 
-  def save_and_render(object, params)
+  def save_and_render(object, params_hash, object_id = nil)
```
Comment (AdamM-FC) [inline app/controllers/application_controller.rb:20]: id is only needed for update/delete so param is optional

File: Gemfile [hunk 2]
```diff
@@ -44,6 +44,8 @@ gem 'pg' , '~> 1.2.3'
 gem 'ruby-kafka', '~> 0.5.4'
 gem 'rubocop', '~> 0.91'
 gem 'solargraph', '~> 0.39.15'
+gem 'avro', '~> 1.8.1'
```
Comment (mugx-fc) [inline Gemfile:47]: you don't need this line, do you?

File: app/kafka/transformer_topic_consumer.rb [hunk 3]
```diff
@@ -5,31 +5,35 @@ def initialize
     Thread.new { subscribe }
   end
 
+  def avro
```
Comment (mugx-fc) [inline app/kafka/transformer_topic_consumer.rb:8]: you can remove this function, it's not used

**Review by mugx-fc — CHANGES_REQUESTED** [review]

File: Gemfile (hunk 2 above)
Comment (AdamM-FC) [reply Gemfile:47]: Oh yeah, can remove this!

File: app/kafka/transformer_topic_consumer.rb (hunk 3 above)
Comment (AdamM-FC) [reply app/kafka/transformer_topic_consumer.rb:8]: Removed

**Review by mugx-fc — APPROVED** [review]
lgtm 👍

**Pull request merged**